File: minegame-database/src/main/java/top/nipuru/minegame/database/processor/QueryPlayerHandler.java
package top.nipuru.minegame.database.processor;

import top.nipuru.minegame.common.message.database.FieldMessage;
import top.nipuru.minegame.common.message.database.QueryPlayerRequest;
import top.nipuru.minegame.common.processor.RequestDispatcher;
import top.nipuru.minegame.database.DatabaseServer;

import java.util.*;

public class QueryPlayerHandler implements RequestDispatcher.Handler<QueryPlayerRequest> {

    private final DatabaseServer server;


    public QueryPlayerHandler(DatabaseServer server) {
        this.server = server;
    }

    @Override
    public void handle(RequestDispatcher.ResponseSender responseSender, QueryPlayerRequest request) throws Exception {
        Map<String, List<List<FieldMessage>>> data = server.getPlayerDataManager().queryPlayer(request);
        responseSender.sendResponse(data);
    }

    @Override
    public Class<QueryPlayerRequest> interest() {
        return QueryPlayerRequest.class;
    }
}

File: minegame-database/src/main/kotlin/top/nipuru/minegame/database/processor/QueryPlayerHandler.kt
package top.nipuru.minegame.database.processor

import com.alipay.remoting.AsyncContext
import top.nipuru.minegame.common.message.database.FieldMessage
import top.nipuru.minegame.common.message.database.QueryPlayerRequest
import top.nipuru.minegame.common.processor.RequestDispatcher
import top.nipuru.minegame.database.player.PlayerDataManager

class QueryPlayerHandler : RequestDispatcher.Handler<QueryPlayerRequest> {

    override fun handle(asyncCtx: AsyncContext, request: QueryPlayerRequest) {
        val data: Map<String, List<List<FieldMessage>>> = PlayerDataManager.queryPlayer(request)
        asyncCtx.sendResponse(data)
    }

    override fun interest(): Class<QueryPlayerRequest> {
        return QueryPlayerRequest::class.java
    }
}
